### PR TITLE
Use rubocop settings inherit from the meta gem

### PR DESCRIPTION
**Pull request by emaglio**
Here what we can do to have the same settings shared for all gems:

1. Having a `rubocop.yml` file where we have the general settings (PR: https://github.com/trailblazer/meta/pull/2)

2. Inherit if using the URL (raw version)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -38,6 +38,8 @@ else # use rubygems releases
   gem "reform-rails"
 end
 
+gem "meta", path: "../meta"
```
Comment (emaglio) [inline Gemfile]: This will go ones meta has a `rubocop.yml` file

File: trailblazer.gemspec [hunk 2]
```diff
@@ -26,6 +26,7 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "bundler"
   spec.add_development_dependency "rake"
+  # spec.add_development_dependency "meta"
```
Comment (emaglio) [inline trailblazer.gemspec]: This will be added ones has a `rubocop.yml` file

**Comment by timoschilling** [comment]
I believe this would be a better way:
```yaml
inherit_from:
  - https://github.com/emaglio/meta/blob/7ee7e01a23937279a9cda9b206994908efce9779/rubocop.yml
  # - https://github.com/trailblazer/meta/blob/master/rubocop.yml # if merged
```

**Comment by emaglio** [comment]
Tried but didn’t work...had an error in YAML.safe_load

**Comment by timoschilling** [comment]
I have never used it, but the rubocop docs telling that it should work.

**Comment by emaglio** [comment]
@timoschilling this is the error generated when using the URL:
```
emaglio@emaglio:~/projects/trailblazer$ rubocop
(/home/emaglio/projects/trailblazer/.rubocop-https---github-com-emaglio-meta-blob-7ee7e01a23937279a9cda9b206994908efce9779-rubocop-yml): mapping values are not allowed in this context at line 318 column 24
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/2.5.0/psych.rb:402:in `parse'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/2.5.0/psych.rb:402:in `parse_stream'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/2.5.0/psych.rb:350:in `parse'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/2.5.0/psych.rb:314:in `safe_load'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader.rb:185:in `yaml_safe_load'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader.rb:158:in `load_yaml_configuration'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader.rb:40:in `load_file'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader_resolver.rb:135:in `block in base_configs'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader_resolver.rb:134:in `map'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader_resolver.rb:134:in `base_configs'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader_resolver.rb:22:in `resolve_inheritance'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader.rb:49:in `load_file'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_loader.rb:83:in `configuration_from_file'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/config_store.rb:44:in `for'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/cli.rb:188:in `apply_default_formatter'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/lib/rubocop/cli.rb:40:in `run'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/bin/rubocop:13:in `block in <top (required)>'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/home/emaglio/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.56.0/bin/rubocop:12:in `<top (required)>'
/home/emaglio/.rbenv/versions/2.5.0/bin/rubocop:23:in `load'
/home/emaglio/.rbenv/versions/2.5.0/bin/rubocop:23:in `<main>'
```
Where line 318 is this:
image

If I remove that line just to see if that is the only issue I got the error to line 472 where the problem is `:`
image

**Comment by timoschilling** [comment]
try it with this url `https://raw.githubusercontent.com/emaglio/meta/7ee7e01a23937279a9cda9b206994908efce9779/rubocop.yml`

**Comment by emaglio** [comment]
Yep the raw version works...didn't know you could have something like that! I'll update this PR then :+1:

File: .rubocop.yml [hunk 3]
```diff
@@ -1,4 +1,6 @@
-inherit_from: .rubocop_todo.yml
+inherit_from:
+    - https://raw.githubusercontent.com/emaglio/meta/7ee7e01a23937279a9cda9b206994908efce9779/rubocop.yml
```
Comment (emaglio) [inline .rubocop.yml]: After merging the PR in `meta` we can use `trailblazer/meta`

**Comment by timoschilling** [comment]
you should update your first comment

**Comment by emaglio** [comment]
Updated the URL using `trailblazer/meta/master`... I guess if we are happy with this we can merge this?

**Pull request merged**